helix-view/src/tree.rs

```diff
@@ -526,6 +526,24 @@ impl Tree {
         }
     }
 
+    pub fn swap_split_in_direction(&mut self, direction: Direction) {
+        if let Some(id) = self.find_split_in_direction(self.focus, direction) {
+            if let Some([focused, target]) = self.nodes.get_disjoint_mut([self.focus, id]) {
+                match (&mut focused.content, &mut target.content) {
+                    (Content::View(focused), Content::View(target)) => {
+                        std::mem::swap(&mut focused.doc, &mut target.doc);
+                        std::mem::swap(&mut focused.id, &mut target.id);
```

Comment:
I'm not particularly fond of swapping the `ViewId`s as well, but the `Document` has it's selection mappings which refers to the `ViewId`. It would require the `Editor` side to always go to the documents effected in the swap and update the keys for the selections... Doing the swap is cheaper, but I'm not sure if it would cause issues in the long run, so I leave it up for reviewers to decide.